var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -280,11 +281,32 @@ def executables(self):
     def filter_detected_exes(cls, prefix, exes_in_prefix):
         result = []
         for exe in exes_in_prefix:
+            # On systems like Ubuntu we might get multiple executables
+            # with the string "gcc" in them. See:
+            # https://helpmanual.io/packages/apt/gcc/
+            #
             # clang++ matches g++ -> clan[g++]
-            if any(x in exe for x in ('clang', 'ranlib')):
+            basename = os.path.basename(exe)
+            substring_to_be_filtered = [
+                'c99-gcc', 'c89-gcc', '-nm', '-ar', 'clang', 'ranlib'
```

Comment:
Can you break this out into a line-by-line list and attach the comment about clang to the line with clang?

```
[
c99-gcc,
-nm,
clang, # clang++ matches g++
...
```